### ISSUE TITLE
New command `check-platform-reqs`

### ISSUE THREAD

**Issue by bkrukowski** [composer/composer#6668]
Would be nice to add new command for composer for checking platform requirements (PHP version and extensions), e.g.:

```
composer check-platform-reqs
```

## Why

I am big fan of sending to production server one zip file with all dependencies instead of executing `composer update --no-dev` on production. Such process takes less time, is [safer](https://medium.com/@azerbike/i-ve-just-liberated-my-modules-9045c06be67c) and can be done offline.

## Usage

New command should be executed on production server after unzip command.

**Comment by stof** [comment]
and I suggest that this command always checks the actual platform, without loading platform overrides.

**Comment by swumao** [comment]
@Seldaek I am interested in this feature. But what should this command output? Now my implement output is below:
```
ext-openssl     7.0.20    requires  ext-openssl (^7.1.0)  failed
ext-pcre        7.0.20    requires  ext-pcre (*)          success
php             7.0.20    requires  php (^5.3.2 || ^7.0)  success
ext-tokenizer   7.0.20    requires  ext-tokenizer (*)     success
ext-dom         20031129  requires  ext-dom (*)           success
ext-json        1.4.0     requires  ext-json (*)          success
ext-Reflection  7.0.20    requires  ext-reflection (*)    success
ext-SPL         7.0.20    requires  ext-spl (*)           success
```

How about this ? Or if you have better advice, please tell me~

thx

**Comment by Seldaek** [comment]
Fixed by #6709
